api/tests/unit/domain/scope-admin/usecases/add-admin-role_test.js

```diff
@@ -0,0 +1,36 @@
+const { expect, sinon } = require('../../../../test-helper');
+const addAdminRole = require('../../../../../lib/domain/scope-admin/usecases/add-admin-role');
+const User = require('../../../../../lib/domain/scope-admin/models/User');
+const { ROLES } = require('../../../../../lib/domain/constants').PIX_ADMIN;
+
+describe('Unit | UseCase | save-admin-member', function () {
+  it('add the role the user', async function () {
+    const parameters = { email: 'ice.bot@example.net', role: ROLES.SUPER_ADMIN };
+    const user = new User({
+      userId: 1,
+      id: 1,
+      firstName: 'John',
+      lastName: 'Doe',
+      email: 'king.k@theboss.com',
+      role: ROLES.SUPPORT,
+      disabledAt: null,
+    });
+    const expectedUser = new User({
+      userId: 1,
+      id: 1,
+      firstName: 'John',
+      lastName: 'Doe',
+      email: 'king.k@theboss.com',
+      role: ROLES.SUPER_ADMIN,
+      disabledAt: null,
+    });
+    const userRepository = { getByEmail: sinon.stub(), save: sinon.stub() };
+    userRepository.getByEmail.resolves(user);
+
+    const actualUser = await addAdminRole({ ...parameters, userRepository });
+
+    expect(actualUser).deep.equal(expectedUser);
+    expect(userRepository.getByEmail).to.have.been.calledWith(parameters.email);
+    expect(userRepository.save).to.have.been.calledWith(expectedUser);
```

Comment:
**question** : du coup, on va préférer avoir des `to.have.been.calledWith` plutôt qu'avoir des `userRepository.getByEmail.withArgs(...).resolves` ?
Avec le withArgs, moins de choses sont à vérifier à la fin, puisque si tu obtiens le `actualUser` alors c'est que les repo ont été appelés avec les bons paramètres.

Ici ça ressemblerait à ça :
```
    const userRepository = { getByEmail: sinon.stub(), save: sinon.stub() };
    userRepository.getByEmail.withArgs(parameters.email).resolves(user);

    const actualUser = await addAdminRole({ ...parameters, userRepository });

    expect(actualUser).deep.equal(expectedUser);
    expect(userRepository.save).to.have.been.calledWith(expectedUser);
```
On ne peut pas faire de même avec le save car il ne retourne rien, et on se récupère pas son retour, mais a un effet de bord sur le modèle du domaine.

Reply:
On pourrait, mais je trouve cette approche plus adaptée au cas ou code fait passe-plat.
C'est pas le cas ici puisqu'on a de la logique métier qui est portée par notre objet métier.
On se contente pas de passer le user récupéré getByEmail pour le passer au save.
Du coup je préfère notre approche.
Même si j'y vois une différence subtile je suis pas sur que standardiser ce test soit apporte beaucoup de valeur.